main/acle.md

```diff
@@ -7766,6 +7920,1419 @@ This is because the ACLE intrinsic calls do not imply a particular
+> it is valid (though pointless) to call it from a function that has
+> [ZA state](#za-state). The compiler would simply reenable ZA after the
+> call and reload the saved ZA state.
+
+### SME ZA state assertions
+
+#### Prototypes
+
+``` c
+  __attribute__((arm_streaming_compatible, arm_shared_za))
+  void svundef_za();
+```
+
+#### Semantics
+
+**`svundef_za()`**
+
+> This call asserts that ZA does not currently have any useful data.
+> Semantically, it fills ZA with unpredictable data, although from
+> a quality of implementation perspective, it should not generate
+> any object code.
+>
+> The call can be used as an optimization hint to the compiler, so that
+> the compiler does not insert unnecessary code to save and restore the
+> current ZA contents. The call might also be useful for static analysis.
+
+### SME instruction intrinsics
+
+#### Common rules
+
+The intrinsics in this section have the following properties in common:
+
+*   Every argument named `tile`, `slice_offset` or `tile_mask` must
+    be an integer constant expression.
+
+*   ZA loads and stores do not use typed pointers, since there is
+    no C or C++ type information associated with the contents of ZA.
+
+*   Intrinsics have a `_hor` suffix if they operate on horizontal slices
+    of a given ZA tile and a `_ver` suffix if they operate on vertical
+    slices of a given ZA tile.
+
+#### LD1B, LD1H, LD1W, LD1D, LD1Q
+
+``` c
+  // Also for _za16, _za32, _za64 and _za128 (with the same prototype).
+  __attribute__((arm_streaming, arm_shared_za))
+  void svld1_hor_za8(uint64_t tile, uint32_t slice_base,
+                     uint64_t slice_offset, svbool_t pg, const void *ptr);
+
+  // Synthetic intrinsic: adds vnum * svcntsb() to the address given by ptr.
+  // Also for _za16, _za32, _za64 and _za128 (with the same prototype).
+  __attribute__((arm_streaming, arm_shared_za))
+  void svld1_hor_vnum_za8(uint64_t tile, uint32_t slice_base,
+                          uint64_t slice_offset, svbool_t pg,
+                          const void *ptr, int64_t vnum);
+
+  // Also for _za16, _za32, _za64 and _za128 (with the same prototype).
+  __attribute__((arm_streaming, arm_shared_za))
+  void svld1_ver_za8(uint64_t tile, uint32_t slice_base,
+                     uint64_t slice_offset, svbool_t pg, const void *ptr);
+
+  // Synthetic intrinsic: adds vnum * svcntsb() to the address given by ptr.
+  // Also for _za16, _za32, _za64 and _za128 (with the same prototype).
+  __attribute__((arm_streaming, arm_shared_za))
+  void svld1_ver_vnum_za8(uint64_t tile, uint32_t slice_base,
+                          uint64_t slice_offset, svbool_t pg,
+                          const void *ptr, int64_t vnum);
+```
+
+#### LDR
+
+``` c
+  // slice_offset fills the role of the usual vnum parameter.
+  __attribute__((arm_streaming_compatible, arm_shared_za))
+  void svldr_vnum_za(uint32_t slice_base, uint64_t slice_offset,
+                     svbool_t pn, const void *ptr);
+```
```

Comment:
The `LDR` instruction in the ISA is unpredicated. I am curious why this ACLE intrinsic uses a predicate?

Reply:
Gah, this a pasto from the LD1 functions. Thanks for the catch.